app/controllers/crops_controller.rb

```diff
@@ -4,40 +4,22 @@ class CropsController < ApplicationController
   before_action :authenticate_member!, except: [:index, :hierarchy, :search, :show]
   load_and_authorize_resource
   skip_authorize_resource only: [:hierarchy, :search]
+  respond_to :html, :json, :rss, :csv
+  responders :flash
 
-  # GET /crops
-  # GET /crops.json
   def index
     @sort = params[:sort]
-    @crops = if @sort == 'alpha'
-               Crop.includes(:scientific_names, plantings: :photos)
-             else
-               popular_crops
-             end
-    @paginated_crops = @crops.approved.paginate(page: params[:page])
-
-    @has_requested_pending = Crop.pending_approval.where(requester: current_member).count if current_member
-
-    respond_to do |format|
-      format.html
-      format.json { render json: @crops }
-      format.rss do
-        @crops = Crop.recent.includes(:scientific_names, :creator)
-        render rss: @crops
-      end
-      format.csv do
-        @filename = "Growstuff-Crops-#{Time.zone.now.to_s(:number)}.csv"
```

Comment:
the filename? It's in line 14 now. telling rails to allow csv is handled in line 7